collector/collector_test.go

```diff
@@ -0,0 +1,234 @@
+		t.Fatal("failed to register collector:", err)
+	}
+
+	jsonData := `{
+	    "TestDetails": {
+	        "TestName": "My Homepage",
+	        "TypeId": "0",
+	        "MonitorTypeId": "11",
+	        "TestId": "123456",
+	        "ReportWindow": "123123123210000000",
+	        "NodeId": "12345",
+	        "NodeName": "Bangalore, IN - Tata Teleservices",
+	        "Asn": "12345",
+	        "DivisionId": "1234",
+	        "ClientId": "123"
+	    },
+	    "Summary": {
+	        "Timestamp": "20240502212044798",
+	        "TotalTime": "",
+	        "Connect": "",
+	        "Dns": "",
+	        "ContentLoad": "",
+	        "Load": "",
+	        "Redirect": "",
+	        "SSL": "",
+	        "Wait": "",
+	        "Client": "",
+	        "DocumentComplete": "",
+	        "RenderStart": "",
+	        "ResponseContent": "",
+	        "ResponseHeaders": "",
+	        "TotalContent": "",
+	        "TotalHeaders": "",
+	        "AnyError": "",
+	        "ConnectionError": "",
+	        "DNSError": "",
+	        "LoadError": "",
+	        "TimeoutError": "",
+	        "TransactionError": "",
+			"ErrorObjectsLoaded": "",
+	        "ImageContentType": "",
+	        "ScriptContentType": "",
+	        "HTMLContentType": "",
+	        "CSSContentType": "",
+	        "FontContentType": "",
+	        "MediaContentType": "",
+	        "XMLContentType": "",
+	        "OtherContentType": "",
+	        "ConnectionsCount": "",
+	        "HostsCount": "",
+	        "FailedRequestsCount": "",
+	        "RequestsCount": "",
+	        "RedirectionsCount": "",
+	        "CachedCount": "",
+	        "ImageCount": "",
+	        "ScriptCount": "",
+	        "HTMLCount": "",
+	        "CSSCount": "",
+	        "FontCount": "",
+	        "XMLCount": "",
+	        "MediaCount": "",
+	        "TracepointsCount": ""
+	    }
+	}`
+	req := httptest.NewRequest("POST", "http://example.com/webhook", strings.NewReader(jsonData))
+	w := httptest.NewRecorder()
+	httpHandler := promhttp.HandlerFor(registry, promhttp.HandlerOpts{})
+	collector.HandleWebhook(w, req)
+
+	// Serve HTTP to trigger metric collection
+	httpHandler.ServeHTTP(w, req)
+
+	// Gather all metrics and assert the count
+	metrics, err := registry.Gather()
+	if err != nil {
+		t.Fatalf("gathering metrics failed: %v", err)
+	}
+
+	// Define expected metric count
+	expectedMetricCount := 1 // Assuming 1 'up' metric for the collector
+	if len(metrics) != expectedMetricCount {
+		t.Errorf("expected %d metrics, got %d", expectedMetricCount, len(metrics))
+	}
+
+	// Read expected metrics from file
+	expectedFile, err := os.Open(filepath.Join("testdata", "empty_metrics.prom"))
+	if err != nil {
+		t.Fatalf("failed to open expected metrics file: %v", err)
+	}
+	defer expectedFile.Close()
+
+	// Compare gathered metrics against expected metrics
+	if err := testutil.GatherAndCompare(registry, expectedFile); err != nil {
+		t.Errorf("gathered metrics did not match expected metrics: %v", err)
```

Comment:
Will it process correctly if some of the metrics are sent empty and others are populated?

Reply:
Yeah it will process if some metrics are empty and some are populated. I added a partial test to show this.